src/k8s-engine/images/imageProcessor.ts

```diff
@@ -61,46 +61,76 @@ export abstract class ImageProcessor extends EventEmitter {
   private refreshInterval: ReturnType<typeof timers.setInterval> | null = null;
   protected images:imageType[] = [];
   protected _isReady = false;
-  private isK8sReady = false;
+  protected isK8sReady = false;
   private hasImageListeners = false;
   private isWatching = false;
   _refreshImages: () => Promise<void>;
   protected currentNamespace = 'default';
+  protected activated = false;
+  protected label = '';
 
-  constructor(k8sManager: K8s.KubernetesBackend) {
+  get name() {
+    return this.label;
+  }
+
+  protected constructor(k8sManager: K8s.KubernetesBackend, label: string) {
     super();
     this.k8sManager = k8sManager;
+    this.label = label;
     this._refreshImages = this.refreshImages.bind(this);
     this.on('newListener', (event: string | symbol) => {
+      if (!this.activated) {
```

Comment:
I think it would make more sense to see if the events on the imageProcessors can be done on the imageEventHandler instead. I put off making that change because this PR was largely about replacing a single image-processor with a set of objects that each implement an image-processor, and this is obviously an area of pain.